shuup/admin/modules/media/views.py

```diff
@@ -130,33 +186,39 @@ def post(self, request, *args, **kwargs):
             return JsonResponse({"error": "unknown action %s" % action})
 
     def handle_get_folders(self, data):
-        root_folders = get_cached_trees(Folder._tree_manager.all())
+        shop = get_shop(self.request)
+        root_folders = get_cached_trees(Folder._tree_manager.filter(_get_folder_query_filter(shop)))
         return JsonResponse({"rootFolder": _filer_folder_to_json_dict(None, root_folders)})
 
     def handle_post_new_folder(self, data):
+        shop = get_shop(self.request)
         parent_id = int(data.get("parent", 0))
         if parent_id > 0:
-            parent = Folder.objects.get(pk=parent_id)
+            parent = _get_folder_query(shop).get(pk=parent_id)
```

Comment:
this could raise `DoesNotExist`, make sure to handle it properly

Reply:
If these untranslated "not found" errors starts to happen then let's add the translation separately. These are catched at post already.